lib/ResourceProviders.ts

```diff
@@ -17,11 +18,17 @@ export class ResourceProviders {
       this.providers[subSet] = [];
     }
 
+    let deps: any[] = Reflect.getMetadata("design:paramtypes", resource)
```

Comment:
Yes, it's not necessary, It's just WebStrom complaining... I have added `import 'reflect-metadata';` into `Resource.ts` and it was enough for ts compiler.